tests/test_get_openapi.py

```diff
@@ -0,0 +1,23 @@
+from fastapi import FastAPI
+from pydantic import BaseModel
+
+
+def test_get_openapi():
+    app = FastAPI()
+
+    class Model(BaseModel):
+        pass
+
+    class Model2(BaseModel):
+        a: Model
+
+    class Model3(BaseModel):
+        c: Model
+        d: Model2
+
+    @app.get("/", response_model=Model3)
+    def f():
+        pass
+
+    openapi = app.openapi()
```

Comment:
The test will fail at this line if the `definitions.update(m_definitions)` is not removed above.